pages/03.Howtos/15.repartition-vm-disk/docs.en.md

```diff
@@ -0,0 +1,255 @@
+---
+title: 'How to repartition VM disk'
+date: '21-10-2020 16:27'
+taxonomy:
+    category:
+        - docs
+---
+
+## How to repartition VM disk
+
+### Overview
+
+This Document will show you the essential steps to repartition your VM disk. For this purpose we are using an Ubuntu 18.04 server as example with the `m1c.tiny` [flavor](../../04.Reference/03.compute/docs.en.md).
+
+
+### Prerequisites
+
+* You need to have the login data for the SysEleven Stack API (user name and passphrase)
+* Knowledge how to utilise a terminal/SSH and SSH-keys.
+
+### Automatically configure partitions
+
+By default [cloudinit](https://cloudinit.readthedocs.io/en/latest/) will grow your VM image partition size to the complete size of the ephemeral disk (which is defined by the flavor you used for your VM). To avoid this behavior and use the spare space for a new partition we will use following snippet :
+
+```shell
+#cloud-config
+growpart:
+  mode: off
+runcmd:
+# Parted asks whether to fix /dev/vda to use all available space, we reply "Fix" and continue
+# Resize primary partition to 10GB
+- "echo Fix > /tmp.txt"
+- "echo 1 >> /tmp.txt"
+- "echo yes >> /tmp.txt"
+- "echo 10GB >> /tmp.txt"
+- "parted ---pretend-input-tty /dev/vda resizepart < /tmp.txt"
+# Prepare partitions
+- "parted -s /dev/vda mkpart DATA ext4 10GB 20GB"
+- "parted -s /dev/vda mkpart DATA ext4 20GB 50GB"
+# Configure new partitions
+- "echo $(parted /dev/vda -s print|grep DATA |awk '/^\ *[0-9]+/{print $1}') > /partition_nums.txt"
+- "for i in $(cat /partition_nums.txt); do mkfs.ext4 /dev/vda${i}; done"
+- "for i in $(cat /partition_nums.txt); do e2label /dev/vda${i} DATA${i}; done"
+- "for i in $(cat /partition_nums.txt); do mkdir -p /mnt/disks/data${i}; done"
+- "for i in $(cat /partition_nums.txt); do mount -t ext4 /dev/vda${i} /mnt/disks/data${i}; done"
+- "for i in $(cat /partition_nums.txt); do echo LABEL=DATA${i} /mnt/disks/data${i} ext4 defaults 0 0 | sudo tee -a /etc/fstab; done"
+# System is not aware of new space on main partition. Needs refresh.
+- "resize2fs /dev/vda1"
+```
+
+This snippet was written for a Ubuntu 18.04 VM with a 50GB ephemeral disk. It will resize the main partition to 10 GB and create 2 new partitions (10 GB and 30 GB), create an ext-4 filesystem and mount them.
+
+Using the OpenStack dashboard to start your VM you may provide this snippet in the `Customization Script` box located in the `Configuration` tab. If you prefer to use the CLI to bring up the VM, you can use the `--user-data` option to provide the cloud-config file containing the snippet.
+
```

Comment:
I suppose you can specify it in terraform as well? Maybe also for heat, since we're supporting that too.

Reply:
Sure, but that should go into the examples rather than the documentation. We could, however, mention those examples here, once they are ready.

Reply:
I agree with @mbernhardtSys11 here. We can modify the page once we created a few examples (e.g. heat and terraform ones).